pkg/scheduler/framework/interface.go

```diff
@@ -160,6 +162,33 @@ func (s *Status) AsError() error {
 	return errors.New(s.Message())
 }
 
+// IsStatusEqual returns whether two Statuses are equal to each other.
+func IsStatusEqual(statusX, statusY *Status) bool {
+	if statusX == statusY {
+		return true
+	}
+	if statusX == nil {
+		return statusY == nil || statusY.code == Success
+	}
+	if statusY == nil {
+		return statusX.code == Success
+	}
+	if statusX.code != statusY.code {
+		return false
+	}
+	if statusX.err == nil && statusY.err == nil {
```

Comment:
why we need to check err == nil?

Reply:
We cannot directly compare reasons for error statuses for wrap cases.